### PR TITLE
[DRAFT] Add zod validator

### PR DESCRIPTION
**Pull request by mnvsk97**
## Changes proposed

- [X] Add an initializer ` in the config folder to validate the environment variables
- [ ] Replace `process.env` with `env.` ensuring only the variables parsed by zod are available
- [ ] Add tests for the initializer

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Note to reviewers

There is an open issue with the compatibility of zod library with playwright. To make sure that the playwright tests pass, I've added a change as suggested [here](https://github.com/microsoft/playwright/issues/22688)

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/6891"></a>

### CODE REVIEW

**Comment by dan-mba** [comment]
I modified `logger.js` to work. Here is the code:
```javascript
import * as PinoLogger from "pino";
import * as Pretty from "pino-pretty"
import env from '@config/env';

let logger;
console.log(env.NODE_ENV)
logger = PinoLogger.pino(
  env.NODE_ENV === "development" ? Pretty({colorize: true}) : {}
);


export default logger;
```

File: config/env.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { z } from "zod";
+import * as dotenv from 'dotenv';
+
+// Load environment variables from .env file
+dotenv.config()
```
[the remark this answers is not in the document]
Comment (mnvsk97) [reply config/env.js]: removed in the latest commits

File: .github/scripts/testimonial.mjs [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import fs from 'fs';
+import env from '@config/env'
```
Comment (dan-mba) [inline .github/scripts/testimonial.mjs:2]: Don't modify this file.
This is used in an action not the LinkFree app

File: pages/playground.js [hunk 3]
```diff
@@ -6,9 +6,10 @@ import Button from "@components/Button";
 import Modal from "@components/Modal";
 import Input from "@components/form/Input";
 import UserPage from "@components/user/UserPage";
+import env from '@config/env'
 
 export default function Playground() {
-  const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
+  const BASE_URL = env.NEXT_PUBLIC_BASE_URL;
```
Comment (dan-mba) [inline pages/playground.js:12]: Can't change this as it is executed on the fron-end

Comment (mnvsk97) [reply pages/playground.js:12]: Didn't get you. What is the difference when this gets executed on the front end? Ultimately, the file will be loaded right?

Comment (dan-mba) [reply pages/playground.js:12]: The logger code will not work on the front end

**Comment by SaraJaoude** [comment]
I will be close this PR due to lack of activity.

**Pull request closed without merging**